src/shims/foreign_items.rs

```diff
@@ -124,10 +212,12 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     ) -> InterpResult<'tcx, Option<&'mir mir::Body<'tcx>>> {
         let this = self.eval_context_mut();
         let attrs = this.tcx.get_attrs(def_id);
-        let link_name = match this.tcx.sess.first_attr_value_str_by_name(&attrs, sym::link_name) {
-            Some(name) => name.as_str(),
-            None => this.tcx.item_name(def_id).as_str(),
+        let link_name_sym = match this.tcx.sess.first_attr_value_str_by_name(&attrs, sym::link_name)
+        {
```

Comment:
Woah these are strange braces... what is rustfmt doing?^^

Maybe using `unwrap_or_else` looks better?